py/bitbox02/bitbox02/util.py

```diff
@@ -39,3 +46,97 @@ def parse_xpub(xpub: str) -> common.XPub:
         chain_code=chain_code,
         public_key=pubkey,
     )
+
+
+class UserCache:
+    """Data structure to hold keys"""
+
+    # pylint: disable=too-few-public-methods
+    def __init__(self, raw_cache: Optional[str] = None):
+        if raw_cache is None:
+            self.app_static_privkey = None
+            self.device_static_pubkeys: List[bytes] = []
+            return
+        (privkey, pubkeys) = UserCache.deserialize(raw_cache)
+        self.app_static_privkey = privkey
+        self.device_static_pubkeys = pubkeys
+
+    def serialize(self) -> str:
+        """Serialize struct to string"""
+        pubkeys = [binascii.hexlify(x).decode("utf-8") for x in self.device_static_pubkeys]
+        privkey = None
+        if self.app_static_privkey is not None:
+            privkey = binascii.hexlify(self.app_static_privkey).decode("utf-8")
+        return json.dumps({"device_static_pubkeys": pubkeys, "app_static_privkey": privkey})
+
+    @staticmethod
+    def deserialize(raw: str) -> Tuple[Optional[bytes], List[bytes]]:
+        """Deserialize content from disk to struct"""
+        try:
+            data = json.loads(raw)
+            privkey = None
+            if data["app_static_privkey"] is not None:
```

Comment:
Should be `if data.get("app_static_privkey") is not None`

Reply:
Actually no, the key will always be there, but the value is None in case it hasn't been set yet

Reply:
I don't see where you are guaranteed to set the key before, what am I missing?

Reply:
https://github.com/digitalbitbox/bitbox02-firmware/pull/354/files#diff-5c7d5e4282d5df35e114d9b5af36ed86R70

It is always serialized with a `privkey`. Either it is `None` or it is `bytes(...)`

Reply:
Mmh, but the class initializer calls `deserialize` first, before serialize has a chance to be called.

Reply:
Right, but if this: `json.loads(raw)` works, then there was a valid json object in the string. And if there was a valid json object then it has that key. I guess my point is that if anyone else puts a valid json file with the wrong content in that location it would throw an exception here yes. I'll update it to catch `KeyError` which would also resolve that issue.